src/components/Tool/FixEmailAddresses/FixEmailAddressPerson/FixEmailAddressPerson.tsx

```diff
@@ -308,61 +319,36 @@ export const FixEmailAddressPerson: React.FC<FixEmailAddressPersonProps> = ({
-                          <Box
-                            display="flex"
-                            justifyContent="space-between"
-                            className={classes.paddingX}
-                          >
-                            <Box>
-                              <Hidden smUp>
-                                <Typography
-                                  display="inline"
-                                  variant="body2"
-                                  fontWeight="bold"
-                                >
-                                  {t('Source')}:
-                                </Typography>
-                              </Hidden>
-                              <Typography display="inline" variant="body2">
-                                {`${email.source} (${dateFormatShort(
-                                  DateTime.fromISO(email.updatedAt),
-                                  locale,
-                                )})`}
-                              </Typography>
-                            </Box>
-                          </Box>
-                        </Grid>
-
-                        <Grid item xs={6} sm={2} className={classes.paddingB2}>
-                          <Box
-                            display="flex"
-                            justifyContent="center"
-                            className={classes.paddingX}
-                          >
-                            <Typography display="flex" alignItems="center">
-                              {email.isPrimary ? (
-                                <>
-                                  <Hidden smUp>
-                                    <Typography
-                                      display="inline"
-                                      variant="body2"
-                                      fontWeight="bold"
-                                    >
-                                      {t('Source')}:
-                                    </Typography>
-                                  </Hidden>
-                                  <StarIcon
-                                    data-testid={`starIcon-${id}-${index}`}
-                                    className={classes.hoverHighlight}
-                                    onClick={() =>
-                                      handleChangePrimary(id, index)
-                                    }
-                                  />
-                                </>
-                              ) : (
-                                <>
+                      <Formik
+                        key={index}
+                        enableReinitialize={true}
+                        initialValues={{
+                          newEmail: email.email,
+                        }}
+                        validationSchema={validationSchema}
+                        onSubmit={async (values) => {
+                          await handleChange(id, index, values.newEmail);
```

Comment:
`handleChange` doesn't return a promise.

```suggestion
                        onSubmit={(values) => {
                          handleChange(id, index, values.newEmail);
```